plaidml/edsl/edsl_test.cc

```diff
@@ -46,6 +46,49 @@ Tensor Softmax(const Tensor& X) {
   return E / N;
 }
 
+TEST(CppEdsl, LowerPrecisionInvalidNegative) {
+  auto A = Placeholder(DType::FLOAT32, {3, 3});
+  auto C = A * (-2);
+
+  EXPECT_ANY_THROW({ Program("lower_precision_constants", {C}, DType::FLOAT16, DType::UINT16); });
+}
+
+TEST(CppEdsl, LowerPrecisionConstants) {
+  auto A = Placeholder(DType::FLOAT32, {3, 3});
```

Comment:
Why would we want to use a FLOAT32 as an input if we are specifying `FLOAT16` for `floatx`?

Reply:
You're right, this isn't really a good example. I've replaced it with a better one